### DISCUSSION TITLE
Stepping into instrumentation in IntelliJ

### DISCUSSION THREAD

**Discussion by jtayl222** [open-telemetry/opentelemetry-java-instrumentation#8832] [Q&A]
When using IntelliJ to debug my application that is instrumented with OpenTelemetry (-javaagent:opentelemetry-javaagent.jar -Dotel.javaagent.configuration-file=agent.properties) I'd like to be able to step into and through OpenTelemetry source. While I *can* see the call stack, the corresponding java source file is not displayed.

Here is an example of the top several stack frames:

get:64, LazyStorage (io.opentelemetry.javaagent.shaded.io.opentelemetry.context)
get:72, ContextStorage (io.opentelemetry.javaagent.shaded.io.opentelemetry.context)
root:104, Context (io.opentelemetry.javaagent.shaded.io.opentelemetry.context)
current:92, Context (io.opentelemetry.javaagent.shaded.io.opentelemetry.context)
currentContext:23, Java8BytecodeBridge (io.opentelemetry.javaagent.bootstrap)

What can I do to get IntelliJ to show the OpenTelemetry source file in an IntelliJ editor pane while debugging?

**Comment by laurit** [comment]
See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/debugging.md#shadow-renaming

**Reply by jtayl222** [reply]
Thanks for the recommendation. We haven't successfully been able to build https://github.com/open-telemetry/opentelemetry-java without the shadow entries, yet. We will get back to you when we can provide more feedback.

**Comment by will-zdu** [comment]
@jtayl222 @laurit   Do I need a collector to debug Instrumentation?

**Reply by laurit** [reply]
no, you don't
